packages/peregrine/lib/talons/CheckoutPage/__tests__/useCheckoutPage.spec.js

```diff
@@ -243,50 +245,49 @@ test('returned error prop should be error from place order mutation', () => {
     expect(talonProps.error).toBeInstanceOf(CheckoutError);
 });
 
-describe('handlePlaceOrder', () => {
-    test('should get order details and place order', async () => {
-        useCartContext.mockReturnValueOnce([
-            { cartId: '123' },
-            { createCart: () => {}, removeCart: () => {} }
-        ]);
-
-        const { talonProps } = getTalonProps(props);
+test('should get order details when handlePlaceOrder called', () => {
+    useCartContext.mockReturnValueOnce([
+        { cartId: '123' },
+        { createCart: () => {}, removeCart: () => {} }
+    ]);
 
-        await talonProps.handlePlaceOrder();
+    const { talonProps } = getTalonProps(props);
 
-        expect(getOrderDetails).toHaveBeenCalledWith({
-            variables: { cartId: '123' }
-        });
-        expect(placeOrder).toHaveBeenCalledWith({
-            variables: { cartId: '123' }
-        });
+    act(() => {
+        talonProps.handlePlaceOrder();
     });
 
-    test('should remove and create new cart', async () => {
-        const createCart = jest.fn();
-        const removeCart = jest.fn();
-        const fetchCartId = jest.fn();
-        useCartContext.mockReturnValueOnce([
-            { cartId: '123' },
-            { createCart, removeCart }
-        ]);
-        createCartMutationResult.mockReturnValue([fetchCartId]);
+    expect(getOrderDetails).toHaveBeenCalledWith({
+        variables: { cartId: '123' }
+    });
+});
 
-        const { talonProps } = getTalonProps(props);
+test("should place order and cleanup when we have order details and place order hasn't been called yet", async () => {
+    const createCart = jest.fn();
+    const removeCart = jest.fn();
+    const fetchCartId = jest.fn();
 
-        await talonProps.handlePlaceOrder();
+    useCartContext.mockReturnValueOnce([
+        { cartId: '123' },
+        { createCart, removeCart }
+    ]);
+    createCartMutationResult.mockReturnValue([fetchCartId]);
 
-        expect(removeCart).toHaveBeenCalled();
-        expect(createCart).toHaveBeenCalledWith({ fetchCartId });
+    useLazyQuery.mockImplementation(() => {
+        return [jest.fn(), { data: {}, loading: false }];
     });
 
-    test('should clear cart data from cache', async () => {
-        const { talonProps } = getTalonProps(props);
+    const { talonProps } = getTalonProps(props);
 
-        await talonProps.handlePlaceOrder();
+    await talonProps.handlePlaceOrder();
 
-        expect(clearCartDataFromCache).toHaveBeenCalledWith(client);
+    expect(placeOrder).toHaveBeenCalledWith({
+        variables: { cartId: '123' }
     });
+    expect(removeCart).toHaveBeenCalled();
+    console.log(clearCartDataFromCache);
+    expect(clearCartDataFromCache).toHaveBeenCalled(); //toHaveBeenCalledWith(client);
```

Comment:
Is this comment relevant?

Reply:
I've cleaned this up - must have forgotten to push.

Reply:
```suggestion
    expect(clearCartDataFromCache).toHaveBeenCalled();
```